### DISCUSSION TITLE
Exporting question

### DISCUSSION THREAD

**Discussion by sbxchicago** [ruffle-rs/ruffle#15650] [Q&A] [answered]
When using the exporter I"m getting the first frame of the swf repeated for as many frames as I request. The swf runs fine in the desktop player. Any help would be greatly appreciated.

Here's the command I'm using:
`cargo run --release --package=exporter -- path-to.swf path-to/screenshots --frames 20`

And here's what I'm seeing in the terminal when I run the command:
```
 Finished release [optimized] target(s) in 1.41s
 Running `target/release/exporter '/Users/brad/Downloads/bhae_animations 3/media/swf/3.2.39.swf' '/Users/brad/Downloads/bhae_animations 3/media/swf/z_screenshots/3.2.39' --frames 20`
[00:00:00] ######################################## [00:00:00]      20/20      Saved first 20 frames of /Users/brad/Downloads/bhae_animations 3/media/swf/3.2.39.swf to /Users/brad/Downloads/bhae_animations 3/media/swf/z_screenshots/3.2.39
```

Thanks

**Comment by n0samu** [comment]
Hi, are you able to share the SWF that you are having an issue with?
If your goal is to render your animation as a video, another option is to use Swivel: https://www.newgrounds.com/wiki/creator-resources/flash-resources/swivel

**Reply by sbxchicago** [reply]
For sure, here's the swf:
http://sandboxstaging.net/3.2.39.swf

Unfortunately, Swivel's not working on my OS (mac ventura).

**Reply by n0samu** [reply, marked as answer]
The animation is contained in a movie clip within the main SWF that only plays after a user interaction (clicking the "Play" button). Automated tools can't really handle this, but you can export the movie clip to a separate SWF using JPEXS Decompiler, then run Ruffle's exporter on that SWF. Or you can export the frames of the clip directly from JPEXS, though that method seems surprisingly slow. Here is a screen recording demonstrating how to do this:

https://github.com/ruffle-rs/ruffle/assets/71368227/a9bf7757-3664-4aa0-8601-0b93a0f1211a

You can select PNG or AVI from the dropdown (instead of SWF) if you want to export the frames directly from JPEXS.

**Reply by sbxchicago** [reply]
Thanks this is incredibly helpful!
